src/network/tarcap/threadpool/packets_blocking_mask.cpp

```diff
@@ -1,90 +1,178 @@
 #include "packets_blocking_mask.hpp"
 
+#include "dag/dag_block.hpp"
+
 namespace taraxa::network::tarcap {
 
-void PacketsBlockingMask::markPacketAsHardBlocked(PriorityQueuePacketType packet_type) {
-  // blocked_packets_peers_time_ contains peers_time blocks that are not compatible with hard blocks, we should
-  // not try to create hard block (and later unblock) when there is ongoing peers_time block
-  assert(blocked_packets_peers_time_.find(packet_type) == blocked_packets_peers_time_.end());
+void PacketsBlockingMask::markPacketAsHardBlocked(const PacketData& blocking_packet,
+                                                  SubprotocolPacketType packet_type_to_block) {
+  auto& packet_hard_block = hard_blocked_packet_types_[packet_type_to_block];
+
+  // If this assert is triggered, it means we are trying to insert blocking packet id that is already inserted, which
+  // should never happen as packets id's are supposed to be unique
+  assert(packet_hard_block.insert(blocking_packet.id_).second);
```

Comment:
This will be removed by release build, huge bug

Reply:
kind of agree, but do you realize that we have tens of asserts throughout the whole codebase with basically the same bug as here ? So all of them are basically huge bugs and we should not use asserts at all

Reply:
Didn't get the comment at first, you are right !